tiatoolbox/models/dataset/info.py

```diff
@@ -0,0 +1,129 @@
+# ***** BEGIN GPL LICENSE BLOCK *****
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# The Original Code is Copyright (C) 2021, TIALab, University of Warwick
+# All rights reserved.
+# ***** END GPL LICENSE BLOCK *****
+
+
+import os
+from abc import ABC, abstractmethod
+
+from tiatoolbox import rcParam
+from tiatoolbox.utils.misc import download_data, grab_files_from_dir, unzip_data
+
+
+class DatasetInfoABC(ABC):
+    """Define an abstract class for holding a dataset information.
+
+    Enforcing such that following attributes must always be defined by the subclass.
+
+    Attributes
+        inputs (list): A list of paths where each path points to a sample image.
+        labels (list): A list of `int` where each is the label of the sample at
+            the same index.
+        label_name (dict): A dict indicates the possible associate name of each label
+            value.
+
+    """
+
+    @property
+    @abstractmethod
+    def inputs(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def labels(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def label_name(self):
+        raise NotImplementedError
+
+
+class KatherPatchDataset(DatasetInfoABC):
+    """Define a class for holding the Kather dataset information.
+
+    Args:
+        save_dir_path (str or None): Path to directory containing the Kather dataset,
+            assumed to be as is after extracted. If the argument is `None`,
+            the dataset will be downloaded and extracted into the
+            'run_dir/download/Kather'.
+
+    Attributes
+        inputs (list): A list of paths where each path points to a sample image.
+        labels (list): A list of `int` where each is the label of the sample at
+            the same index.
+        label_name (dict): A dict indicates the possible associate name of each label
+            value.
+
+    """
+
+    # We predefine to follow enforcement, actual initialization in init
+    inputs = None
+    labels = None
+    label_name = None
+
+    def __init__(
+        self,
+        save_dir_path=None,
+    ):
+        label_codes = [
```

Comment:
Suggest 'label_name' instead of 'label_codes' as it's easier to guess what it is from the name.